app/ide-desktop/lib/dashboard/src/components/dashboard/AssetRow.tsx

```diff
@@ -758,7 +757,7 @@ export default function AssetRow(props: AssetRowProps) {
                   }
                 }}
                 className={tailwindMerge.twMerge(
-                  'h-row rounded-full transition-all ease-in-out rounded-rows-child',
+                  'h-table-row rounded-full transition-all ease-in-out rounded-rows-child',
```

Comment:
fair enough. IIRC i explicitly set this one because it wasn't filling the full height (noticeable when you select multiple rows.) i'll take a look whether it can be solved in a nicer way

Reply:
actually on second thought: if the height is not provided here, then where will it come from?

Reply:
What's the problem with inheriting it from the height of the nested elements?

Reply:
we explicitly want a specific height for the rows, especially so that it's consistent between regular, loading, and "this directory is empty"/"you have no projects" rows. it's certainly possible to hack it to work by giving the nested elements a specific margin, but again i'm not sure it's a good idea as different elements will have different intrinsic heights

Reply:
By design, we _don't want_ a _specific_ height for a table. see: 
![CleanShot 2024-06-17 at 15 04 37@2x](https://github.com/enso-org/enso/assets/61194245/6cfb3559-7986-4c1a-9092-a63032534074)

If you take a closer look at the design, you see that the size of the text perfectly matches the button size on the left.
Items are aligned inside the grid. Components serve that purpose. 

And I want from you the same: instead of trying to stretch the result to the conditions, build the UI properly from the ground up.

Reply:
again it's worth noting that each column has its own maximum height. this would mean that, if the height of the row depends on the height of its contents, then the height of each individual row depends not only on the set of visible columns, but also on whether or not any user has the read/view permissions on this asset.

also note that `h-table-row` is for the height of the row itself, not the button and not the text. if you have a *productive* suggestion then i'm more than happy to apply it, but this advice:

> inheriting it from the height of the nested elements

is not applicable for reasons already stated above.

Reply:
> if the height of the row depends on the height of its contents, then the height of each individual row depends not only on the set of visible columns, but also on whether or not any user has the read/view permissions on this asset.

Might be wrong, but isn't it exactly what we want? And this is exactly how table layouts work, right? In my opinion, it's weird if a row had differently-sized columns. So whats the exact problem?

> itself, not the button and not the text.

This is exactly why it scares me. If we need to make something smaller/larger/add wrapping - 💥

Reply:
again - the issue is that *different* rows in the *same* table will have *different* (= inconsistent) heights, depending on the contents of the columns.

and as for row heights changing based on which columns are visible - this causes layout shift that is otherwise totally avoidable

> if we need to make something smaller/larger/add wrapping - 💥

the idea here is that if there is a situation in which a row is ever larger than this predefined threshold, either:

- change the global height of all rows, once, or
- `min-h-max` or similar

Reply:
> again - the issue is that different rows in the same table will have different (= inconsistent) heights, depending on the contents of the columns.

It's not true, because in design it's not happening. 
You can achieve the same result without hardcoding the height of a row just by paying enough attention to the elements inside the table. But you'll have a much more predictable markup and more robust UI.

Reply:
https://github.com/enso-org/enso/pull/9918/commits/36c21af57962d86099d0e9af26b27d2cb94ebc1e#diff-6fd8d84a5877d3cda838fb03b76e127fbfd3e608534945df1e8ff92f7f51ef1eL147

For example, here, I could just set `h-6` on the button and achieve the same result, but it's incorrect. The proper solution is to _find out why_ the element has the wrong height and fix it

Reply:
an astute observation - except for the fact it *is* explicitly set in the design. (see the "drive" page - the one of the "main" page is actually a reference to the original copy on the "drive" page. actually the 38px height is the *only* explicitly set (white) value)

![image](https://github.com/enso-org/enso/assets/4046547/fc426dce-b4e4-4a99-abe9-7decb9387864)

if you right click the `entry` components and go to the main component you will see that the height of the component itself is actually 32px:

![image](https://github.com/enso-org/enso/assets/4046547/3ea4baf4-75e9-4cea-922c-5577fa4fe68a)

Reply:
As far as I can see it's a workaround in design. This element doesn't require explicit height

Reply:
mind explaining further?